src/riak_core_security.erl

```diff
@@ -82,6 +98,28 @@
 -type bucket() :: {binary(), binary()} | binary().
 -type permission() :: {string()} | {string(), bucket()}.
 -type userlist() :: all | [string()].
+-type username() :: string().
+-type metadata_key() :: string().
+-type metadata_value() :: term().
+-type options() :: [{metadata_key(), metadata_value()}].
+
+-spec find_by_metadata(metadata_key(), metadata_value()) -> [{username(), options()}].
+find_by_metadata(Key, Value) ->
+    riak_core_metadata:fold(
+      fun({_Username, [?TOMBSTONE]}, Acc) ->
```

Comment:
It would be good if we could extract this multi-headed anonymous function into a named fun - would have to have a very small anonymous fun here for the closure over {Key, Value} but otherwise just call, what, `include_user_with_option({_UserName, [_Options]} = Data, Key, Value, Acc)` or something like that?